### DISCUSSION TITLE
Dotfiles customizations

### DISCUSSION THREAD

**Discussion by numbsafari** [community/community#5573] [Codespaces] [answered]
Really loving codespace so far. I've wanted this for so long, it's really awesome to see it happening. Great work pulling it off.

Working with dotfiles, I have a couple of quick thoughts:

1.  It'd be great if it were possible to specify a branch in your public `dotfiles` repo that should be used for CodeSpaces, rather than defaulting to `main`/`master`. This is a common convention for dealing with linux v. macos, home v. work, etc. Would be great to have that flexibility here.

2.  As a manager for some orgs, it would be nice to disallow team members from using their personal `dotfiles` repos, and instead use an org repo where their username maps to a branch. Or let the user specify that as an org-specific setting. This would be helpful for ensuring internal tooling and project names don't accidentally get leaked, and also cut down on random code/dependency injection into your private repositories.

Hopefully I didn't miss these things in the docs.

**Comment by 2percentsilk** [comment, marked as answer]
Thanks for this feedback! You didn't miss anything in the docs and these are all great ideas to improve our dotfiles workflow with Codespaces.

While I can't offer an ETA on when we might start working on this, we'll keep it in mind as we look at our roadmap.
